web/src/message_view.ts

```diff
@@ -1225,14 +1294,19 @@ export function narrow_by_topic(target_id, opts) {
         {operator: "channel", operand: original.stream_id.toString()},
         {operator: "topic", operand: original.topic},
     ];
-    opts = {then_select_id: target_id, ...opts};
-    show(search_terms, opts);
+    show(search_terms, {then_select_id: target_id, ...opts});
 }
 
-export function narrow_by_recipient(target_id, opts) {
-    opts = {then_select_id: target_id, ...opts};
+export function narrow_by_recipient(
+    target_id: number,
+    opts: {
+        trigger: string;
+    },
+): void {
+    const show_opts = {then_select_id: target_id, ...opts};
```

Comment:
This `opts` => show_opts` refactor should be a prep commit too.

Reply:
done